### ISSUE TITLE
tcmalloc calls new on Initialization

### ISSUE THREAD

**Issue by JohnSully** [gperftools/gperftools#1160]
My program overrides the new/delete operators and forwards them to malloc/free after some additional bookkeeping.  This works fine with jemalloc however because TCMalloc calles new as part of it's initialization this causes a recursive loop.

My analysis of the code shows this could be resolved with static object instead of one dynamically allocated.

My questions are:
1) Is there some special method that already exists to solve this use case?  I've seen issues in the past about overriding new/delete so my understanding is this use case is supported.

2) Would a PR changing these dynamic allocations in the initialization path to static objects be accepted?

**Comment by alk** [comment]
No, such use was not anticipated yet. Yes, patch to fix this case, I think is acceptable. Thanks.
